src/main/java/com/openlattice/shuttle/dates/JavaDateTimeHelper.java

```diff
@@ -1,81 +1,123 @@
+            if ( checkDatePatternIsTwoDigitYear( datePattern ) ) {
                 // TODO: break this out into its own transform that specifies the date boundaries for two-digit years
-                if ((ld.getYear() - LocalDate.now().getYear()) > 20) {
-                    ld = ld.withYear(ld.getYear() - 100);
+                if ( ( ld.getYear() - LocalDate.now().getYear() ) > 20 ) {
+                    ld = ld.withYear( ld.getYear() - 100 );
                 }
             }
             return ld;
-        });
+        } );
     }
 
-    public OffsetDateTime parseDateTime(String date) {
-        return parseWithTwoDigitYearHandling( date, ( toParse, formatter ) -> {
-            LocalDateTime parsed = LocalDateTime.parse( toParse, formatter );
-            return parsed.atZone( tz.toZoneId() ).toOffsetDateTime();
-        }, ( odt, datePattern ) -> {
-            if (  checkDatePatternIsTwoDigitYear( datePattern ) ) {
-                // TODO: break this out into its own transform that specifies the date boundaries for two-digit years
-                if ((odt.getYear() - LocalDate.now().getYear()) > 20) {
-                    odt = odt.withYear(odt.getYear() - 100);
-                }
+    public OffsetDateTime parseDateTime( String date ) {
+        try {
+            // DateTime: first try parsing as iso-string into OffsetDateTime
+            // TimeZone: if Timezone is not consistent with reported TimeZone: warning
+            OffsetDateTime odt = OffsetDateTime.parse( date );
+            if ( tz.isPresent() ) {
+                if ( tz.orElse( Constants.DEFAULT_TIMEZONE ).toZoneId() == odt.getOffset() );
+                logger.error( "The reported and requested timezones are inconsistent." );
             }
             return odt;
-        });
+        } catch ( DateTimeParseException | IllegalArgumentException eAutoParseODT ) {
+            try {
+                // DateTime: try parsing iso-string into TimeStamp
+                // TimeZone: set tz to provided tz
+                Timestamp ts = Timestamp.valueOf( date );
+                OffsetDateTime odt = OffsetDateTime.ofInstant( ts.toInstant(), tz.orElse( Constants.DEFAULT_TIMEZONE ).toZoneId() );
+                return odt;
+            } catch ( DateTimeParseException | IllegalArgumentException eAutoParseTs ) {
+                return parseWithTwoDigitYearHandling(
+                        date,
+                        ( toParse, formatter ) -> {
+                            try {
+                                // DateTime: try parsing into ODT with provided patterns
+                                // TimeZone: do not use tz argument
+                                OffsetDateTime odt = OffsetDateTime.parse( toParse, formatter );
+                                if ( tz.isPresent() ) {
+                                    if ( tz.orElse( Constants.DEFAULT_TIMEZONE ).toZoneId() == odt.getOffset() );
```

Comment:
Same comment here as https://github.com/openlattice/shuttle/pull/141/files#r377951899

Reply:
Actually, it should be enough to check tz only once probably at the very beginning of this function

Reply:
I don't think so because the way to deal with the tz differs from case to case.  If there's a timezone in the string/object, then it should be dealt with differently than when none is present.  See the flowchart in PR.

Reply:
`java.lang.IllegalArgumentException: Timestamp format must be yyyy-mm-dd hh:mm:ss[.fffffffff]`

Reply:
Yeah, you are right, I missed that it matches the `OffsetDateTime`, sorry!